velox/functions/prestosql/ArrayFunctions.h

```diff
@@ -728,6 +728,38 @@ struct ArrayRemoveNullFunctionString {
   }
 };
 
+template <typename T>
+struct ArrayNGramFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T)
+
+  void call(
+      out_type<Array<Array<Generic<T1>>>>& out,
+      const arg_type<Array<Generic<T1>>>& input,
+      int64_t n) {
+    if (n <= 0) {
+      VELOX_USER_FAIL("N must be positive.");
```

Comment:
use VELOX_USER_CHECK_GE for better error messages (e.g. n will appear in the message automatically)